core/src/main/java/org/infinispan/persistence/async/AsyncCacheWriter.java

```diff
@@ -115,8 +124,38 @@ public void start() {
       DefaultThreadFactory coordinatorThreadFactory =
             new DefaultThreadFactory(null, Thread.NORM_PRIORITY, DefaultThreadFactory.DEFAULT_PATTERN, nodeName,
                                      "AsyncStoreCoordinator");
-      coordinator = coordinatorThreadFactory.newThread(new AsyncStoreCoordinator());
+
+      coordinator = coordinatorThreadFactory.newThread(new AsyncStoreCoordinator(asyncConfiguration.failSilently()));
       coordinator.start();
+
+      pollStoreAvailability();
+      // If no errors on start, then start observing the status of the underlying store. We monitor the availability of
+      // the delegate cache store here, opposed to the PersistenceManager, as it's possible for AsyncCacheWriter to be
+      // available even if the delegate isn't, this occurs when the modification queue is still able to queue requests
+      long interval = ctx.getCache().getCacheConfiguration().persistence().availabilityInterval();
+      ctx.getExecutor().scheduleAtFixedRate(this::pollStoreAvailability, interval, interval, TimeUnit.MILLISECONDS);
+   }
+
+   private void pollStoreAvailability() {
+      boolean available = false;
+      try {
+         available = actual.isAvailable();
+      } catch (Throwable t) {
+         log.debugf("Error encountered when calling isAvailable on %s: %s", actual, t);
+      }
+      availabilityLock.lock();
+      try {
+         if (available != delegateAvailable) {
+            if (available) {
+               delegateAvailable = true;
```

Comment:
I would say just assign `delegateAvailable = available` and remove the else block.